src/test/java/com/google/devtools/build/lib/query2/testutil/SkyframeQueryHelper.java

```diff
@@ -245,7 +245,8 @@ public AbstractBlazeQueryEnvironment<Target> getQueryEnvironment() {
 
   @Override
   public ResultAndTargets<Target> evaluateQuery(String query)
-      throws QueryException, InterruptedException {
+      throws QueryException, InterruptedException, AbruptExitException {
+    skyframeExecutor.handleDiffsForTesting(getReporter());
```

Comment:
I was thinking whether it's better to do it here, or just do it locally at the tests that concerns the ignored paths. Doing it here is probably fine since it better reflects the behavior of actual Bazel.

The reason for having this here isn't immediately obvious though, so please add a quick comment explaining this. Thanks.